File: sample-event-emitter/src/test/java/nl/haploid/octowight/sample/repository/AtomChangeEventRepositoryIT.java
package nl.haploid.octowight.sample.repository;

import nl.haploid.octowight.sample.AbstractIT;
import nl.haploid.octowight.sample.TestData;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

import static org.junit.Assert.assertEquals;

public class AtomChangeEventRepositoryIT extends AbstractIT {

	@Autowired
	private AtomChangeEventDmoRepository repository;

	@Test
	@Transactional
	public void testFindAll() {
		final AtomChangeEventDmo event = TestData.atomChangeEventDmo();
		repository.saveAndFlush(event);
		final List<AtomChangeEventDmo> events = repository.findAll();
		assertEquals(1, events.size());
	}
}

File: sample-event-emitter/src/test/scala/nl/haploid/octowight/sample/repository/AtomChangeEventRepositoryIT.scala
package nl.haploid.octowight.sample.repository

import nl.haploid.octowight.sample.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class AtomChangeEventRepositoryIT extends AbstractIT {
  @Autowired private[this] val atomChangeEventDmoRepository: AtomChangeEventDmoRepository = null

  "Atom change event repository" should "find all events" in {
    atomChangeEventDmoRepository.deleteAll()
    val event = TestData.atomChangeEventDmo
    atomChangeEventDmoRepository.saveAndFlush(event)
    val events = atomChangeEventDmoRepository.findAll
    events should have size 1
  }
}
